### PR TITLE
Introduce slog package

### LINKED ISSUE #613: Investigate the usage of the new standard slog package
## Is your feature request related to a problem? Please describe

We are using external packages to handle logging but Go has introduced a standard one for structured logs.
Let's investigate how it behaves and also its performance in comparison with the existing packages.

## Describe the solution

Let's investigate.

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Resolves #613.

## Short description of the changes

- Remove all logging packages and introduce slog
- Mandatory upgrade to Go 1.19

### CODE REVIEW

**Review by xenosdio — APPROVED** [review]

File: client/grpc/grpc.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func unaryInterceptor(target string) grpc.UnaryClientInterceptor {
 		carrier := headersCarrier{Ctx: ctx}
 		err := span.Tracer().Inject(span.Context(), opentracing.TextMap, &carrier)
 		if err != nil {
-			log.FromContext(ctx).Errorf("failed to inject tracing headers: %v", err)
+			log.FromContext(ctx).Error("failed to inject tracing headers", slog.Any("error", err))
```
Comment (pkritiotis) [inline client/grpc/grpc.go:77]: Do we still need the log package here for the ctx?

File: log/log.go
Comment (pkritiotis) [inline log/log.go:1]: Isn't it strange that FromContext is not provided by slog?
I don't quite understand how InfoCtx works but could create a new logger with the corresponding attributes and add it to the context and then use InfoCtx with the already populated context?

**Review by pkritiotis — APPROVED** [review]
Mega PR, great upgrade 🚀

File: client/grpc/grpc.go (hunk 1 above)
Comment (mantzas) [reply client/grpc/grpc.go:77]: yes, unfortunately, `slog` does not have a feature to support logging propagation via context...
This allows us to create loggers with extra data attributes per request...

File: log/log.go
Comment (mantzas) [reply log/log.go:1]: Funny story. If you look at the implementation of the handles (json and text) ctx is not even used...
check out the code: https://cs.opensource.google/go/x/exp/+/master:slog/json_handler.go;l=88

**Pull request merged**